tests/phpunit/tests/option/option.php

```diff
@@ -312,4 +312,214 @@ public function data_option_autoloading() {
+			),
+		);
+
+		// Make sure options are not in cache or database initially.
+		$this->assertFalse( wp_cache_get( 'option1', 'options' ) );
+		$this->assertFalse( wp_cache_get( 'option2', 'options' ) );
+		$this->assertFalse( wp_cache_get( 'option3', 'options' ) );
+
+		// Call the prime_options_by_group function to prime the options.
+		prime_options_by_group( 'group1' );
+
+		// Check that options are now in the cache.
+		$this->assertEquals( get_option( 'option1' ), wp_cache_get( 'option1', 'options' ) );
+		$this->assertEquals( get_option( 'option2' ), wp_cache_get( 'option2', 'options' ) );
+
+		// Make sure option3 is still not in cache.
+		$this->assertFalse( wp_cache_get( 'option3', 'options' ) );
+	}
+
+	/**
+	 * Test get_options function.
+	 *
+	 * @ticket 58962
+	 *
+	 * @covers ::prime_options
+	 * @covers ::get_options
+	 */
+	public function test_get_options() {
+		// Create some options to prime.
+		$options_to_prime = array(
+			'option1',
+			'option2',
+		);
+
+		// Make sure options are not in cache or database initially.
+		$this->assertFalse( wp_cache_get( 'option1', 'options' ) );
+		$this->assertFalse( wp_cache_get( 'option2', 'options' ) );
+
+		// Call the prime_options function to prime the options.
+		prime_options( $options_to_prime );
+
+		// Check that options are now in the cache.
+		$this->assertEquals( get_option( 'option1' ), wp_cache_get( 'option1', 'options' ) );
+		$this->assertEquals( get_option( 'option2' ), wp_cache_get( 'option2', 'options' ) );
+
+		// Call the get_options function to retrieve the options.
+		$options = get_options( array( 'option1', 'option2' ) );
+
+		// Check that the retrieved options are correct.
+		$this->assertEquals( get_option( 'option1' ), $options['option1'] );
+		$this->assertEquals( get_option( 'option2' ), $options['option2'] );
+	}
+
+	/**
+	 * Test get_options with an empty input array.
+	 *
+	 * @ticket 58962
+	 *
+	 * @covers ::get_options
+	 */
+	public function test_get_options_with_empty_array() {
+		// Call the get_options function with an empty array.
+		$options = get_options( array() );
+
+		// Make sure the result is an empty array.
+		$this->assertEmpty( $options );
+	}
+
+	/**
+	 * Test prime_options with options that do not exist in the database.
+	 *
+	 * @ticket 58962
+	 *
+	 * @covers ::prime_options
+	 */
+	public function test_prime_options_with_nonexistent_options() {
+		// Create some options to prime.
+		$options_to_prime = array(
+			'option1',
+			'option2',
+		);
+
+		// Make sure options are not in cache or database initially.
+		$this->assertFalse( wp_cache_get( 'option1', 'options' ) );
+		$this->assertFalse( wp_cache_get( 'option2', 'options' ) );
+
+		// Call the prime_options function to prime the options.
+		prime_options( $options_to_prime );
+
+		// Check that options are not in the cache or database.
+		$this->assertFalse( wp_cache_get( 'option1', 'options' ) );
+		$this->assertFalse( wp_cache_get( 'option2', 'options' ) );
```

Comment:
Per the change that @joemcgill suggested, we should test here that the options are now present in the `notoptions` cache (since they don't exist).

Reply:
Yes. In https://github.com/WordPress/wordpress-develop/pull/4970/commits/b191e7b8d719a92c0e1f192c3360333ce5edc340 I address this feedback.